botorch/models/approximate_gp.py

```diff
@@ -157,6 +163,8 @@ def posterior(
         posterior = GPyTorchPosterior(distribution=dist)
         if hasattr(self, "outcome_transform"):
             posterior = self.outcome_transform.untransform_posterior(posterior)
+        if posterior_transform is not None:
+            posterior = posterior_transform(posterior)
```

Comment:
To add test coverage here you could add another check to https://github.com/pytorch/botorch/blob/main/test/models/test_approximate_gp.py#L74 to use a simple posterior transform and make sure it's properly applied